OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -366,32 +375,41 @@ private async Task<SearchResult<ApplicationDto>> GetByProviderId(Guid id, Applic
             return applications;
         }
 
-        private async Task CheckUserRights(Guid parentId = default, Guid providerId = default, Guid workshopId = default)
+        private async Task<IEnumerable<ApplicationDto>> GetByProviderAdminId(string userId, ApplicationFilter filter)
+        {
+            var providerAdmin = await providerAdminService.GetById(userId).ConfigureAwait(false);
+
+            if (providerAdmin is null)
+            {
+                throw new ArgumentException(localizer[$"There is no providerAdmin with userId = {userId}"]);
+            }
+
+            await CheckUserRights(providerAdminId: userId).ConfigureAwait(false);
+
+            var applications = await applicationService.GetAllByProviderAdmin(userId, filter, providerAdmin.ProviderId, providerAdmin.IsDeputy).ConfigureAwait(false);
+
+            return applications;
+        }
+
+        private async Task CheckUserRights(Guid parentId = default, Guid providerId = default, Guid workshopId = default, string providerAdminId = default)
         {
             var userId = GettingUserProperties.GetUserId(User);
 
             bool userHasRights = true;
```

Comment:
not your code, but maybe the default value should be false, it's better to deny access by default in case something goes wrong. Also maybe it's better to refactor the exception flow in case of `userHasRights` is `false` to return a `403` code instead of a more general bad request as it is now.